modules/cfe_testcase/src/resource_id_misc_test.c

```diff
@@ -0,0 +1,161 @@
+}
+
+void TestGetBaseSerial(void)
+{
+    UtPrintf("Testing: CFE_ResourceId_GetBase, CFE_ResourceId_GetSerial");
+    CFE_ES_AppId_t     AppId;
+    CFE_ES_LibId_t     LibId;
+    CFE_ES_TaskId_t    TaskId;
+    CFE_ES_CounterId_t CounterId;
+    CFE_ES_MemHandle_t PoolId;
+    CFE_ES_CDSHandle_t CDSHandleId;
+    const char *       LibName     = "ASSERT_LIB";
+    const char *       CounterName = "TEST_COUNTER";
+    int8               Pool[1024];
+    size_t             CDSBlockSize = 10;
+    const char *       CDSName      = "TEST_CDS";
+
+    /* Referenced from cfe_core_resourceid_basevalues.h */
+    int TASKID_BASE     = CFE_RESOURCEID_MAKE_BASE(OS_OBJECT_TYPE_OS_TASK);
+    int APPID_BASE      = CFE_RESOURCEID_MAKE_BASE(OS_OBJECT_TYPE_USER + 1);
+    int LIBID_BASE      = CFE_RESOURCEID_MAKE_BASE(OS_OBJECT_TYPE_USER + 2);
+    int COUNTID_BASE    = CFE_RESOURCEID_MAKE_BASE(OS_OBJECT_TYPE_USER + 3);
+    int POOLID_BASE     = CFE_RESOURCEID_MAKE_BASE(OS_OBJECT_TYPE_USER + 4);
+    int CDSBLOCKID_BASE = CFE_RESOURCEID_MAKE_BASE(OS_OBJECT_TYPE_USER + 5);
+
+    /* App ID */
+    UtAssert_INT32_EQ(CFE_ES_GetAppID(&AppId), CFE_SUCCESS);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetBase(AppId), APPID_BASE);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetSerial(AppId), AppId - APPID_BASE);
+    /* Lib ID */
+    UtAssert_INT32_EQ(CFE_ES_GetLibIDByName(&LibId, LibName), CFE_SUCCESS);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetBase(LibId), LIBID_BASE);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetSerial(LibId), LibId - LIBID_BASE);
+    /* Task ID */
+    UtAssert_INT32_EQ(CFE_ES_GetTaskID(&TaskId), CFE_SUCCESS);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetBase(TaskId), TASKID_BASE);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetSerial(TaskId), TaskId - TASKID_BASE);
+    /* Counter ID */
+    UtAssert_UINT32_EQ(CFE_ES_RegisterGenCounter(&CounterId, CounterName), CFE_SUCCESS);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetBase(CounterId), COUNTID_BASE);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetSerial(CounterId), CounterId - COUNTID_BASE);
+    /* Pool ID */
+    UtAssert_INT32_EQ(CFE_ES_PoolCreate(&PoolId, Pool, sizeof(Pool)), CFE_SUCCESS);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetBase(PoolId), POOLID_BASE);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetSerial(PoolId), PoolId - POOLID_BASE);
+    /* CDS Block Id */
+    CFE_ES_RegisterCDS(&CDSHandleId, CDSBlockSize, CDSName);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetBase(CDSHandleId), CDSBLOCKID_BASE);
+    UtAssert_INT32_EQ(CFE_ResourceId_GetSerial(CDSHandleId), CDSHandleId - CDSBLOCKID_BASE);
+}
+
+void TestFindNext(void)
+{
+    UtPrintf("Testing: CFE_ResourceId_FindNext");
+    CFE_ES_AppId_t AppId;
+    UtAssert_INT32_EQ(CFE_ES_GetAppID(&AppId), CFE_SUCCESS);
+    /*
+     * To test FindNext better the internal function CFE_ES_CheckCounterIdSlotUsed needs to be reimplemented or accessed
+     * somehow. The current test just uses dummy functions to always return true or false
+     */
```

Comment:
Functional Tests generally try to avoid dummy functions. If they can't be tested without them then usually better to just note that they can't be.

Reply:
@skliper thoughts on what to do here?

Reply:
Done 👍

Reply:
Looks good to me.